lang/ghc/Portfile

```diff
@@ -29,144 +24,173 @@ long_description    \
-# relevant here. This code block is duplicated in this port to avoid
-# inconsistencies.
-compiler.blacklist-append \
-                llvm-gcc-4.2 \
-                gcc-4.2 \
-                {clang < 503.0.38} \
-                macports-clang-3.3
-
-# ghc needs a build and runtime dependency on the compiler used to build it
-# same is also set in ghc-bootstrap. clang-4.0 works, is needed on older systems anyway
-# also /usr/bin/clang does not work on 10.11, so override it there
-if { ${os.platform} eq "darwin" && ( ${os.major} < 13 || ${os.major} == 15 ) } {
-    compiler.whitelist    macports-clang-4.0
-    depends_run-append    port:clang-4.0
+build {
+    ui_msg "\tBuild performed in destroot."
 }
 
-set bootstraproot ${prefix}/share/ghc-bootstrap
-set llvmPrefix  ${prefix}/libexec/llvm-3.5
-configure.args  --with-ghc=${bootstraproot}/bin/ghc \
-                --with-iconv-includes=${prefix}/include \
-                --with-iconv-libraries=${prefix}/lib \
-                --with-gmp-includes=${prefix}/include \
-                --with-gmp-libraries=${prefix}/lib \
-                --with-ffi-includes=${prefix}/lib/libffi-3.2.1/include \
-                --with-ffi-libraries=${prefix}/lib \
-                --with-system-libffi \
-                --with-gcc="${configure.cc}" \
-                --with-llc=${llvmPrefix}/bin/llc \
-                --with-opt=${llvmPrefix}/bin/opt
-
-# OK so because the bootstrap binary has been prebuilt for libraries in
-# /usr/lib we search these in addition to MacPorts' paths to prevent link
-# errors. GHC _should_ actually compile itself in stage2 using paths from the
-# command line arguments
-compiler.cpath /usr/include
-compiler.library_path /usr/lib
-
-build.args      VERBOSE=1
-destroot.args   VERBOSE=1
-
-pre-activate {
-    set obsoletes [list]
-
-    # Legacy port deactivation hack added 2012-12-08, hs-platform-ghc no longer exists
-    lappend obsoletes hs-platform-ghc
-    # Legacy port deactivation hack added 2013-08-11, hs-process is provided by base
-    lappend obsoletes hs-process
-    # Legacy port deactivation hack added 2013-08-14, hs-template-haskell is provided by base
-    lappend obsoletes hs-template-haskell
-    # Legacy port deactivation hack added 2015-01-05, hs-transformers, hs-xhtml
-    # and hs-binary are provided by base
-    lappend obsoletes hs-transformers
-    lappend obsoletes hs-xhtml
-    lappend obsoletes hs-binary
-
-    foreach obsolete $obsoletes {
-        if {![catch {set obsolete_installed [lindex [registry_active $obsolete] 0]}]} {
-            # $obsolete is installed and active
-            # force deactivation
-            registry_deactivate_composite $obsolete "" [list ports_nodepcheck 1]
-        }
+destroot {
+    ui_msg "\tBuild/destroot from ${name}-${version}-x86_64-apple-darwin${extract.suffix}…"
+    system -W ${worksrcpath} "make install -j\$((`/usr/bin/sysctl -n hw.physicalcpu`+1))"
```

Comment:
Our base should provide a variable for that (I believe it does, but I'm not behind a mac at the moment, I could check later).

Reply:
A variable for the number of physical CPUs?

Reply:
I believe there is such a thing; parallel builds are common in MacPorts.

Reply:
```
find /opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/_resources/port1.0/group/ -type f -exec fgrep -i CPU -- {} /dev/null ';'
```

shows that there's a `sysctl` tcl command. I'll use that.

Reply:
Actually, it's just much cleaner to do this in the shell, which is already been invoked. I'm going to keep it as is.

BTW, I've noticed that MacPorts uses `hw.logicalcpu` for its make commands, which uses all resources. A much better option is `hw.physicalcpu`+1 == `hw.logicalcpu`/2+1, as done here.